src/openscm_runner/adapters/magicc7/_run_magicc_parallel.py

```diff
@@ -154,23 +209,23 @@ def run_magicc_parallel(cfgs, output_vars, output_config):
             initializer=_init_magicc_worker,
             initargs=(shared_dict,),
         )
+        try:
+            res = _parallel_process(
+                func=_execute_run,
+                configuration=runs,
+                pool=pool,
+                config_are_kwargs=True,
+                front_serial=2,
+                front_parallel=2,
+            )
+
+            LOGGER.info("Appending results into a single ScmRun")
+            res = scmdata.run_append([r for r in res if r is not None])
+
+        finally:
+            instances.cleanup()
+            LOGGER.info("Shutting down parallel pool")
+            shared_manager.shutdown()
+            pool.shutdown()
 
-        res = _parallel_process(
-            func=_execute_run,
-            configuration=runs,
-            pool=pool,
-            config_are_kwargs=True,
-            front_serial=2,
-            front_parallel=2,
-        )
-
-        LOGGER.info("Appending results into a single ScmRun")
-        res = scmdata.run_append([r for r in res if r is not None])
-
-    finally:
-        instances.cleanup()
-        LOGGER.info("Shutting down parallel pool")
-        shared_manager.shutdown()
-        pool.shutdown()
-
-    return res
+        return res
```

Comment:
It might be clearer to hoist the return into the try block as it will only be hit if no exceptions are raised

Reply:
True, I simplified the return so it returns the final result directly.